### ISSUE TITLE
Method Returns

### ISSUE THREAD

**Issue by taylorjdawson** [corpetty/py-etherscan-api#8]
I have been trying to determine the what data ought to be returned with a method call.
As of right now most methods return only the "result" from the json request but leave of the "status" and "message" portion. I am wondering if the user of the python api should be getting the **whole** request back?
My reasoning is that there should be an error check implemented somewhere. 
For example if you make an erroneous request it returns following:

`{"status":"0","message":"No transactions found","result":[]}`

Should the error check be handled by the python (internally) or by the api user (externally)?

**Comment by taylorjdawson** [comment]
I think I found a nice solution...

I utilized the python equivalent of an anonymous inner class, [type](https://docs.python.org/3/library/functions.html#type), to return an object with the instance variables: status, message, result. This will enable the API user to select which section of the response that they wish to read. It also enables them to check the status of the request placing the burden on the user to handle invalid api responses as does python's built in request client (e.g. `if requests.status_code == 200:`)

For the time being, the subclass will only need a minor revision as shown below:

Current | Revised
------------ | -------------
`req['status']` | `req.status`
`req['message']` | `req.message`
`req['result']` | `req.result`

However, in the future I would like to return the whole object to the user. That would require heavy revisions of the account class. Although I do believe that these changes would be beneficial.

**Comment by corpetty** [comment]
Can you give me an idea of the potential benefits of returning the whole object, and offloading the handling to the user?

I do like the idea of returning a object instance, which lends to more of an OOP style.

**Comment by taylorjdawson** [comment]
- I think that it would look more elegant
- It would also enable those using a Smart IDE to use auto-complete which is handy
- It would allow the user more flexibility. For instance, when performing iterable tasks they could check `if req.status == '1'` which would allow for them to, more conveniently, handle errors in their implementation.

Just an idea!

**Comment by corpetty** [comment]
I think this is a good idea, just because increased generality is better than not.  I don't think the userbase of this package is large enough to impact serious projects either.

**Comment by taylorjdawson** [comment]
Okay great! I'll get to work on the implementation.

**Comment by taylorjdawson** [comment]
@AndreMiras Could I get your input on this?

**Comment by AndreMiras** [comment]
Thanks for asking!
So basically we change the json dictionary-like response to a more object like to access it using `result.attribute` rather than `result['key']` correct? I'm fine with that.
But you also suggested something about giving more context from the response, like `Response.status_code` correct? That I also like, but I doesn't seem to be the case when doing `type('', (object,), json.loads(req.text))` as you suggested, is it?

**Comment by AndreMiras** [comment]
Yes agree to have the entire JSON response. The question is also the `requests.Response` object, that's the one that contains HTTP `status_code`.

**Comment by maiganne** [comment]
hi,when I use the parameter “&to=” to filter the transction ,but the result return is the same ,does not the prrameter work?
